libsolidity/ast/AsmJsonImporter.cpp

```diff
@@ -54,7 +54,7 @@ T AsmJsonImporter::createAsmNode(Json::Value const& _node)
 {
 	T r;
 	r.location = createSourceLocation(_node);
-	astAssert(!r.location.isEmpty() || !r.location.source, "Invalid source location in Asm AST");
```

Comment:
Shouldn't this rather be something like `isValid() || !hasText()`?

Reply:
Checking `AsmJsonImporter::createSourceLocation()` and `langutil::parseSourceLocation()`, it seems we always set `start`, `end`, `source`.
So just `astAssert(r.location.hasText() ....)` might be OK.